multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLoggerPersister.java

```diff
@@ -0,0 +1,61 @@
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.sql.DataSource;
+
+import org.cloudfoundry.multiapps.controller.persistence.model.ImmutableOperationLogEntry;
+import org.cloudfoundry.multiapps.controller.persistence.model.OperationLogEntry;
+import org.cloudfoundry.multiapps.controller.persistence.query.providers.SqlOperationLogQueryProvider;
+import org.springframework.scheduling.annotation.Async;
+
+@Named("processLoggerPersister")
+public class ProcessLoggerPersister {
+
+    private final ProcessLoggerProvider processLoggerProvider;
+    private final DataSource dataSource;
+
+    @Inject
+    public ProcessLoggerPersister(ProcessLoggerProvider processLoggerProvider, DataSource dataSource) {
+        this.processLoggerProvider = processLoggerProvider;
+        this.dataSource = dataSource;
+    }
+
+    @Async("asyncExecutor")
+    public void persistLogs(String correlationId, String activityId) {
+        List<ProcessLogger> processLoggers = processLoggerProvider.getExistingLoggers(correlationId, activityId);
+        Map<String, StringBuilder> processLogsMessages = new HashMap<>();
+
+        for (ProcessLogger processLogger : processLoggers) {
+            if (processLogsMessages.containsKey(processLogger.getOperationLogEntry()
+                                                             .getOperationLogName())) {
+                processLogsMessages.get(processLogger.getOperationLogEntry()
+                                                     .getOperationLogName())
+                                   .append(processLogger.getLogMessage());
+            } else {
+                StringBuilder logMessage = new StringBuilder();
+                logMessage.append(processLogger.getLogMessage());
+                processLogsMessages.put(processLogger.getOperationLogEntry()
+                                                     .getOperationLogName(),
+                                        logMessage);
+            }
+
+            processLoggerProvider.removeProcessLoggerFromCache(processLogger);
+        }
+
+        for (var processLogsMessage : processLogsMessages.entrySet()) {
+            OperationLogEntry operationLogEntry = ImmutableOperationLogEntry.copyOf(processLoggers.get(0)
+                                                                                                  .getOperationLogEntry())
+                                                                            .withOperationLogName(processLogsMessage.getKey())
+                                                                            .withOperationLog(processLogsMessage.toString())
+                                                                            .withModified(LocalDateTime.now());
+            SqlOperationLogQueryProvider.saveLogInDatabase(operationLogEntry, dataSource);
```

Comment:
isn't this logic still generate a lot of sql queries, one for each log?

Reply:
well, no so many because the idea is that in the map are saved logs in the format: operation-log-name: operation-log. operation-log-name can be for example ztana.log so there is one query for the logs that are related to ztana and for every different operation-log-name.